base_ical/models/base_ical.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2023 Hunki Enterprises BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3.0)
+
+
+import datetime
+from urllib.parse import urlparse, urlunparse
+
+import pytz
+import vobject
+from dateutil import relativedelta
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class BaseIcal(models.Model):
+    _name = "base.ical"
+    _description = "Definition of an iCal export"
+
+    active = fields.Boolean(default=True)
+    name = fields.Char(required=True, translate=True)
+    model_id = fields.Many2one("ir.model", required=True, ondelete="cascade")
+    model = fields.Char(related="model_id.model")
+    domain = fields.Char(
+        required=True, default="[]", help="You can use variables `env` and `user` here"
+    )
+    preview = fields.Text(compute="_compute_preview")
+    expression_dtstamp = fields.Char(
+        required=True,
+        vevent_field="dtstamp",
```

Comment:
I am unfamiliar with this capability of Odoo fields. You're able to decorate the field with a custom vevent_field argument?

Reply:
you can add [arbitrary arguments](https://github.com/OCA/OCB/blob/15.0/odoo/fields.py#L289), you should however add it in `_valid_field_parameter` to avoid a [warning](https://github.com/OCA/OCB/blob/15.0/odoo/fields.py#L477). Very handy as this makes the code below much more concise